src/features/session_trace/aggregate/index.js

```diff
@@ -145,6 +148,14 @@ export class Aggregate extends AggregateBase {
     }
     /* --- EoS --- */
 
+    setTimeout(() => {
```

Comment:
Hmm...couldn't we do this in the prepareHarvest method instead of creating a new timeout?

Reply:
Initially tried but it became really complicated because of how many factors are at play here. Like you'd have to consider the "final" or remaining nodes harvest and that `#prepareHarvest` is triggered on regular interval harvests and `subscribeToEOL`. It becomes a lot simpler to separate this logic from having to think of when and how `#prepareHarvest` is called.